privacyidea/lib/security/encryptkey.py

```diff
@@ -0,0 +1,346 @@
+        with open(outfile, "wb") as f:
+            f.write(bytearray(r))
+
+    def _listkeys(self, keytype="public"):
+        if keytype == "public":
+            objs = self.session.findObjects([(PyKCS11.CKA_CLASS, PyKCS11.CKO_PUBLIC_KEY)])
+        else:
+            objs = self.session.findObjects([(PyKCS11.CKA_CLASS, PyKCS11.CKO_PRIVATE_KEY)])
+        log.debug("Found {0!s} keys.".format(len(objs)))
+        for obj in objs:
+            log.debug("========================================================")
+            log.debug("Found object {0!s}".format(obj))
+
+    def _decrypt_file(self, filename):
+        f = open(filename, "rb")
+        filecontents = f.read()
+        f.close()
+        privkey = self._get_private_key()
+        m = PyKCS11.Mechanism(MECHANISM)
+        r = self.session.decrypt(privkey, filecontents, m)
+        r = int_list_to_bytestring(r)
+        for key_id in [0, 1, 2]:
+            self.secrets[key_id] = r[key_id * 32: (key_id + 1) * 32]
+        log.info("Successfully loaded encryption keys into process.")
+
+    def _get_secret(self, slot_id=0, password=None):
+        """
+        This replaces the _get_secret method from the base class to return
+        one of the three encryption keys.
+
+        :param slot_id: The id of the encryption key
+        :param password: n/a
+        :return:
+        """
+        key = self.secrets.get(slot_id)
+        return key
+
+    def setup_module(self, params):
+        """
+
+        :param params:
+        :return:
+        """
+        log.warning("The method 'setupmodule' is not implemented and can not be used.")
+
+
+if __name__ == "__main__":  # pragma: no cover
+    logging.basicConfig()
+    log.setLevel(logging.INFO)
+    log.setLevel(logging.DEBUG)
+
+    try:
+        opts, args = getopt.getopt(sys.argv[1:], "hm:s:k:i:o:e:l",
+                                   ["help", "module=", "slotname=", "keyid=",
+                                    "infile=", "outfile=",
+                                    "encfile=", "listprivate", "listpublic", "keylabel="])
+    except getopt.GetoptError as e:
+        print(str(e))
+        sys.exit(1)
+
+    infile = outfile = None
+    config = {}
+    listkeys = False
+
+    for o, a in opts:
+        if o in ("-m", "--module"):
+            config["module"] = a
+        elif o in ("-s", "--slotname"):
+            config["slotname"] = a
+        elif o in ("-k", "--keyid"):
+            config["keyid"] = int(a)
+        elif o in ("--keylabel"):
+            config["keylabel"] = a
+        elif o in ("-i", "--infile"):
+            infile = a
+        elif o in ("-o", "--outfile"):
+            outfile = a
+        elif o in ("-e", "--encfile"):
+            config["encfile"] = a
+        elif o in ("-l", "--listpublic"):
+            listkeys = "public"
+        elif o in ("-l", "--listprivate"):
+            listkeys = "private"
+
+    p = EncryptKeyHardwareSecurityModule(config, logout=False)
```

Comment:
Currently this bails out with a python error if some configuration parameter is missing. Should we add some descriptions in case of error?

Reply:
Hm, as this will be probably only used three times in two years I do not think so.